core/src/main/java/io/questdb/cutlass/text/CopyTask.java

```diff
@@ -1183,10 +1184,10 @@ private void importPartitionDataVanilla(
                         long nextOffset = nextLengthAndOffset & MASK;
 
                         // line indexing stops on first EOL char, e.g. \r, but it could be followed by \n
-                        int diff = ((int) (nextOffset - offset)) - bytesToRead;
-                        int nextBytesToRead = diff + nextLineLength;
+                        long diff = nextOffset - offset - bytesToRead;
+                        int nextBytesToRead = (int) (diff + nextLineLength);
                         if (diff > -1 && diff < 2 && bytesToRead + nextBytesToRead <= fileBufSize) {
-                            bytesToRead += diff + nextLineLength;
+                            bytesToRead += ((int) diff) + nextLineLength;
```

Comment:
Can be simplified to the following?
```java
bytesToRead += nextBytesToRead;
```

Reply:
it can! I had a slightly different version before, then tweaked it a bit and ended-up with duplicated things without noticing. thanks!